lib/bubbles.js

```diff
@@ -21,23 +21,40 @@ Bubble.prototype.draw = function(context) {
 };
 
 Bubble.prototype.move = function() {
-  if (this.status === "new") {
-    this.x+=this.xInc;
-  } else if (this.status === "floating") {
-    this.y--;
+  var bubble = this;
+  float(bubble);
+  bubble.count++;
+  if (doneFloatingSidewaysOrHitAWall(bubble)) {
+    bubble.status = "floating";
   }
-  this.count++;
-  if (this.x >= (this.canvas.width - 5 ) || this.x <= 0 || this.count === 50) {
-    this.status = "floating";
+  if (onRightEdge(bubble)) {
+    bubble.x = bubble.canvas.width - bubble.width;
```

Comment:
It looks like we handle the case where we're beyond the right wall, but we don't necessarily handle the same issue with the left wall.